src/main/kotlin/io/zerobase/smarttracing/resources/QRCodeGenerator.kt

```diff
@@ -0,0 +1,84 @@
+package io.zerobase.smarttracing.resources
+
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.EncodeHintType
+import com.google.zxing.WriterException
+import com.google.zxing.client.j2se.MatrixToImageWriter
+import com.google.zxing.qrcode.QRCodeWriter
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel
+import java.awt.*
+import java.awt.image.BufferedImage
+import java.io.File
+import java.io.IOException
+import java.util.*
+import javax.imageio.ImageIO
+
+class GenerateQRCode(var logoPath: String, // have to use source path
```

Comment:
naming nit: this is an action. class names should be nouns: `QrCodeGenerator` would be better

Reply:
nit: have the file name the same name as the class name